File: BackEnd/MicroServices/InternshipOffer/src/main/java/com/example/internshipoffer/Service/CompanyService.java
package com.example.internshipoffer.Service;

import com.example.internshipoffer.Entity.Company;

import java.util.List;
import java.util.Optional;

public interface CompanyService {
    Company addCompany(Company company);
    List<Company> getAllCompanies();
    Optional<Company> getCompanyById(Long id);
    Company updateCompany(Long id, Company companyDetails);
    void deleteCompany(Long id);
}

File: FrontEnd/src/app/services/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Company } from '../models/Company';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {
  private apiUrl = 'http://localhost:8083/companies';
  constructor(private http: HttpClient) {}

  getCompanies(): Observable<Company[]> {
    return this.http.get<Company[]>(this.apiUrl);
  }

  getCompanyById(id: number): Observable<Company> {
    return this.http.get<Company>(`${this.apiUrl}/${id}`);
  }

  addCompany(company: Company): Observable<Company> {
    return this.http.post<Company>(this.apiUrl, company);
  }

  updateCompany(id: number, company: Company): Observable<Company> {
    return this.http.put<Company>(`${this.apiUrl}/${id}`, company);
  }

  deleteCompany(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
